### ISSUE TITLE
Make new training from un-finished pretrained model

### ISSUE THREAD

**Issue by iumyx2612** [ultralytics/yolov5#5942]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

Supposed I have an old un-finished training model with **m/M** epochs done.
I stopped it because I don't feel the model can get any better. 
I update my dataset, pick up the weight of that model to continue training with it.
`!python train.py --weights=runs/train/ppe_yolov5m_hyp.scratch/weights/best.pt --data=data/ppe2.yaml --name=ppe_datasetv2_v5m_pretrained_cutout --batch-size=32 --epochs=N
`
The new training starts from epoch **m+1** instead of 0, and the best.pt isn't getting update in the new training folder even if I get a best result in this **new training run**
So I want to load the model state_dict and architecture only, not including `ckpt['optimizer']` and `ckpt['best_fitness']` and `ckpt['ema']`

### Use case

Quit training early, and can still make a new training with only weights from an un-finished training model

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@iumyx2612 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="LR Curves" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

## Start from Partially Trained

Run `utils.general.strip_optimizer()` on your partially trained `last.pt` or `best.pt` to finalize it's training, and then use it like any other pretrained weights:
```bash
python -c "from utils.general import *; strip_optimizer('path/to/best.pt')"  # strip
python train.py --weights path/to/best.pt  # start new training
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by iumyx2612** [comment]
> @iumyx2612 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/9f3a388cea66ca71b9bbe057929eed8fbc920a7e/train.py#L109-L142) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.
> 
> <img alt="LR Curves" width="850" src="https://user-images.githubusercontent.com/26833433/119332195-8e113a80-bc88-11eb-8eb7-7494ff1d5c01.png">
> 
> If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:
> 
> ## Resume Single-GPU
> You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
> 
> ```shell
> python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
> python train.py --resume path/to/last.pt  # specify resume checkpoint
> ```
> 
> ## Resume Multi-GPU
> [Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
> 
> ```shell
> python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
> python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
> ```
> 
> ## Start from Pretrained
> If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
> 
> ```shell
> python train.py --weights path/to/best.pt  # start from pretrained model
> ```
> 
> ## Start from Partially Trained
> Run `utils.general.strip_optimizer()` on your partially trained `last.pt` or `best.pt` to finalize it's training, and then use it like any other pretrained weights:
> 
> ```shell
> python -c "from utils.general import *; strip_optimizer('path/to/best.pt')"  # strip
> python train.py --weights path/to/best.pt  # start new training
> ```
> 
> Good luck 🍀 and let us know if you have any other questions!

Correct me if I'm wrong
`utils.general.strip_optimizer()` will discard the `ckpt['optimizer']`, `ckpt['training_result']`,  `ckpt['wandb_id']`,  `ckpt['ema']`,  `ckpt['updates']` [(line 741)](https://github.com/ultralytics/yolov5/blob/master/utils/general.py#L741)
but when saving the `ckpt` dict, there's no `ckpt['training_result']`, instead it's `ckpt['best_fitness']` ([line 337](https://github.com/ultralytics/yolov5/blob/master/train.py#L377))
I haven't run the script yet but is this a mistake? If not, can you explain this to me? I think the  `ckpt['best_fitness']` should also be discarded when called by `utils.general.strip_optimizer()`

**Comment by glenn-jocher** [comment]
@iumyx2612 ah, yes you are correct! The two list of keys appear to have diverged. Can you please submit a PR to update utils.general.strip_optimizer()? Thank you!
